icaruscode/CRT/CRTTPCTruthEffMCKalman_module.cc

```diff
@@ -0,0 +1,1721 @@
+
+		crt_hitflag = closest.thishit.ts0_s_corr;
+
+		//extract all info about CRT Hit's mac5+channel+pes into vectors
+	    	std::map< uint8_t, std::vector<std::pair<int,float> > > crthit_pes = closest.thishit.pesmap; ///< Saves signal hit information (FEB, local-channel and PE) .
+		for(uint8_t i=1; i<232; i++){
+			if(crthit_pes[i].size()>0){
+				for(int j=0; j<(int)crthit_pes[i].size(); j++){
+					matched_hit_mac5.push_back(i);
+					matched_hit_chan.push_back(crthit_pes[i][j].first);
+					matched_hit_pes.push_back(crthit_pes[i][j].second);
+				}//end for(int j=0; j<crt_pes[i].size(); j++)
+			}//end if(crthit_pes[i].size()>0)
+		}//end loop searching for FEBs connected to the hit
+
+ 		std::vector<icarus::match_geometry> all_crt_candidates = t0Alg.GetClosestCRTHit_geo(detProp, *trackList[track_i], hits, crtHits, m_trigger_timestamp, fIsData);
+
+			auto checkhit = closest.thishit;
+			crt_trueID =  bt.TrueIdFromTotalEnergy(event, checkhit);
+			GetAncestorID(crt_trueID, crt_mother, crt_ancestor, crt_motherlayers, particles);
+			crt_pdg = particles[crt_trueID].PdgCode();
+			crt_ancestor_pdg = particles[crt_ancestor].PdgCode();
+
+		has_crtmatch=false;
+		num_crt_candidates = (int)all_crt_candidates.size();
+		if(num_crt_candidates>0){
+			has_crtmatch=true;
+			for(int i=0; i<num_crt_candidates; i++){
+
+				auto thiscand = all_crt_candidates[i];
+				sbn::crt::CRTHit crtcandidate = thiscand.thishit;
+		
+					int temp_crt_trueid=INT_MAX, temp_crt_motherid=INT_MAX, temp_crt_ancestorid=INT_MAX, temp_crt_motherlayers=INT_MAX;
+					temp_crt_trueid = bt.TrueIdFromTotalEnergy(event, crtcandidate);
+				 	GetAncestorID(temp_crt_trueid, temp_crt_motherid, temp_crt_ancestorid, temp_crt_motherlayers, particles);
+					all_crt_pdg.push_back(particles[temp_crt_trueid].PdgCode()); 
+	
+					all_crt_candidate_trueIDs.push_back(temp_crt_trueid);
+					all_crt_candidate_motherIDs.push_back(temp_crt_motherid);
+					all_crt_candidate_ancestorIDs.push_back(temp_crt_ancestorid);
+					all_crt_candidate_motherlayers.push_back(temp_crt_motherlayers);
+
+
+				all_crt_candidate_x.push_back(thiscand.crt_hit_pos.X());
+				all_crt_candidate_y.push_back(thiscand.crt_hit_pos.Y());
+				all_crt_candidate_z.push_back(thiscand.crt_hit_pos.Z());
+
+				all_crt_regions.push_back((int)fCrtutils->AuxDetRegionNameToNum(thiscand.thishit.tagger));
+				crt_timestamp.push_back(thiscand.crtTime);
+
+				crt_start_dca.push_back(thiscand.simpleDCA_startDir);	
+				crt_end_dca.push_back(thiscand.simpleDCA_endDir);	
+
+				all_crt_hitcode_map.push_back(i);//Gives ID to each hit for the FEB+channel+pes breakdown below to allow cross-referencing between vectors
+
+				all_crt_hitflags.push_back(thiscand.thishit.ts0_s_corr);
+
+			    	std::map< uint8_t, std::vector<std::pair<int,float> > > thishit_pes = crtcandidate.pesmap; ///< Saves signal hit information (FEB, local-channel and PE) .
+				for(uint8_t k=1; k<232; k++){
+					if(thishit_pes[k].size()>0){
```

Comment:
```suggestion
					if (thishit_pes[k].size() == 0) continue;
```
Similar to the comment above about reducing nesting.